include/zephyr/mgmt/ec_host_cmd/backend.h

```diff
@@ -21,98 +21,111 @@
+	 * The handler takes @a handler_owns to read data in @a buf.
+	 */
+	struct k_sem handler_owns;
 };
 
 /**
- * @brief Context for host command backend and framework to pass tx data
+ * @brief Context for host command backend and handler to pass tx data
  */
 struct ec_host_cmd_tx_buf {
-	/** Data to write to the host */
+	/** Data to write to the host The buffer is provided by the handler if
+	 * EC_HOST_CMD_HANDLER_TX_BUFFER > 0. Otherwise, the backend should provide
+	 * the buffer on its own and overwrites @a buf pointer and @a len_max
+	 * in the init function.
+	 */
 	void *buf;
 	/** Number of bytes to write from @a buf */
 	size_t len;
+	/** Size of @a buf */
+	size_t len_max;
 };
 
-typedef int (*ec_host_cmd_backend_api_init)(
-	const struct device *dev, struct ec_host_cmd_rx_ctx *rx_ctx);
-
-typedef int (*ec_host_cmd_backend_api_send)(
-	const struct device *dev,
-	const struct ec_host_cmd_tx_buf *tx_buf);
-
-__subsystem struct ec_host_cmd_backend_api {
-	ec_host_cmd_backend_api_init init;
-	ec_host_cmd_backend_api_send send;
+/**
+ * This structure is common to all backends and is expected to be
+ * the first element in a backend-specific config structure.
+ */
+struct ec_host_cmd_config {
+	/* Backend device instance */
+	const struct device *dev;
 };
 
 /**
- * @brief Initialize a host command device
+ * @brief Initialize a host command backend
  *
- * This routine initializes a host command device, prior to its first use. The
- * receive context object are an output of this function and are valid
- * for the lifetime of this device. The RX context is used by the client to
- * receive data from the host.
+ * This routine initializes a host command backend. It includes initialization
+ * a device used to communication and setting up buffers.
+ * This function is called by the ec_host_cmd_init function. However, if
+ * DT_CHOSEN(zephyr_host_cmd_backend) is defined, the ec_host_cmd_init function
+ * should be called automatically by a backend using SYS_INIT with
+ * CONFIG_EC_HOST_CMD_INIT_PRIORITY priority.
  *
- * @param dev Pointer to the device structure for the driver instance.
- * @param rx_ctx [out] The receiving context object that are valid for the
- *               lifetime of the device. These objects are used to receive data
- *               from the driver when the host send data.
+ * @param backend [in] A pointer to the backend structure for the driver instance.
+ * @param config  [in] A pointer to backend dependent config data
+ * @param rx_ctx  [in,out] A pointer to the receive context object.
+ *		  These objects are used to receive data from the driver
+ *		  when the host sends data. The buf member can be assigned by the
+ *		  backend.
+ * @param tx      [in,out] A pointer to the transmit buffer object. The buf and
+ *		  len_max members can be assigned by the backend. These objects
+ *		  are used to send data by the backend with the
+ *		  ec_host_cmd_backend_api_send function.
  *
  * @retval 0 if successful
```

Comment:
These look a bit funky in the generated documentation, I think because you changed the indentation. You should align the "0 if successful" with the other entries so that the doc renders as "Return values <break> 0 if successful" rather than "Retval 0: <break> if successful" (see https://builds.zephyrproject.io/zephyr/pr/54132/docs/services/device_mgmt/ec_host_cmd.html#c.ec_host_cmd_backend_api_init)

Reply:
Oh nevermind then. The link: click on "Documentation Build / Documentation Build (HTML)"  details -> gear icon -> "view job summary"